---
title: C28105
ms.date: 11/04/2016
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 455e06476d6841f468a64d45b350e45f3e8d1a75
ms.sourcegitcommit: 42ea834b446ac65c679fa1043f853bea5f1c9c95
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 04/19/2018
---
# <a name="c28105"></a>C28105
Warnung C28105: Speicherverlust Ressource aufgrund einer Ausnahme

 Die angegebene Ressource ist nicht freigegeben, wenn eine Ausnahme ausgelöst wird. Die Anweisung, die vom Pfad angegebenen kann eine Ausnahme ausgelöst. Diese Warnung ist ähnlich wie in "Warnung" [C28103](../code-quality/c28103.md), außer dass in diesem Fall eine Ausnahme beteiligt ist.

## <a name="example"></a>Beispiel
 Im folgenden Codebeispiel wird diese Warnung generiert:

```
res = KeSaveFloatingPointState(buffer);

res = AllocateResource(Resource);
char *p2 = new char[10]; // could throw

delete[] p2;
FreeResource(Resource)
```

 Im folgenden Codebeispiel wird diese Warnung vermieden:

```
res = AllocateResource(Resource);
char *p2;

try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
FreeResource(Resource)
```